data/containers/Dockerfile.sle

```
@@ -0,0 +1,19 @@
+FROM baseimage_var
+#registry.suse.de/suse/sle-15-sp3/update/cr/totest/images/suse/sle15:15.3
+
+RUN zypper -n in python3 python3-pip curl
```

With this simple adjustment you can rename this file back to `Dockerfile`:
```suggestion
RUN if [ -x "$(command -v zypper)" ] && [ ! -x "$(command -v pip3)" ]; then zypper -n in python3 python3-pip curl; fi
```

---

@pdostal lets discuss this. the reason is that i was thinking that it would be impossible to maintain one Dockerfile for everything. Considering that we might want to add other things in the future. this is why i chose to not touch the existing Dockerfile but to add a new one. i am not sure i want to start having conditions in any Dockerfiles.

---

I get your point and I agree that in case the file starts to be more complex or starts to server multiple purposes then it makes sense to separate it. In this case tho we still have simple python app utilising only `pip` and that is why I would keep one file. Let's maybe wait until Monday and ask the other as I may be wrong.

---

I'd vote against this change and leave `RUN zypper -n in python3 python3-pip curl` until we need to change this. Is there a reason we want to skip this installation on a Docker container, that is SLE specific? I don't think we will ever need to have this condition there in the first place, as this always runs on SLE?
Also: If `zypper` or `pip3` are not available, I doubt the rest of the Dockerfile will run without errors.

I see the intention here but am afraid that this might be only half of the solution for a problem that will probably never arise, right?

---

> I'd vote against this change and leave `RUN zypper -n in python3 python3-pip curl` until we need to change this. Is there a reason we want to skip this installation on a Docker container, that is SLE specific? I don't think we will ever need to have this condition there in the first place, as this always runs on SLE?

This dockerfile has similarities with the one that it used by another job. That other Dockerfile uses a base image that it already includes python and etc. The request was to unify them. 

> Also: If `zypper` or `pip3` are not available, I doubt the rest of the Dockerfile will run without errors.
> 
> I see the intention here but am afraid that this might be only half of the solution for a problem that will probably never arise, right?

---

Pavel just explained it to me. Since this should be also used outside of SLE then the approach is fine. Find then the filename a bit misleading, but very minor remark.

Feel free to proceed :+1:

---

> This dockerfile has similarities with the one that it used by another job. That other Dockerfile uses a base image that it already includes python and etc. The request was to unify them.

The `Dockerfile` has no similarities, it's simple a copy. Both files are building one simple Python app.
To reply to @grisu48, we wanna also build this `FROM python:3` where `zypper` should be skipped.

I'd also rename it back from `Dockerfile.sle` to just `Dockerfile`.

---

Ok, from my side it's a OK then. Both approaches should be fine and it depends really on how far you want to go.

---

We do not want to go any further. We only want to build this simple app.

---

file renamed to Dockerfile.test. the rest as we discussed offline